pennylane/ops/op_math/sum.py

```diff
@@ -331,30 +332,80 @@ def arithmetic_depth(self) -> int:
         return 1 + max(summand.arithmetic_depth for summand in self.summands)
 
     @classmethod
-    def _simplify_summands(cls, sum_op: "Sum") -> List[Operator]:
-        """Reduces the depth of nested summands.
+    def _simplify_summands(cls, summands: List[Operator]):
+        """Reduces the depth of nested summands and groups equal terms together.
 
-        If ``depth`` is not provided or negative, then the summands list is completely flattenned.
-
-        Keyword Args:
-            depth (int): Reduced depth. Default is -1.
+        Args:
+            summands (List[~.operation.Operator]): summands list to simplify
 
         Returns:
-            List[~.operation.Operator]: reduced summands list
+            .SumSummandsGrouping: Class containing the simplified and grouped summands.
         """
-        summands = []
-        for summand in sum_op.summands:
+        new_summands = SumSummandsGrouping()
+        for summand in summands:
+            # This code block is not needed but it speeds things up when having a lot of  stacked Sums
             if isinstance(summand, Sum):
-                summands.extend(cls._simplify_summands(sum_op=summand))
+                sum_summands = cls._simplify_summands(summands=summand.summands)
+                for op_hash, [coeff, sum_summand] in sum_summands.queue.items():
+                    new_summands.add(summand=sum_summand, coeff=coeff, op_hash=op_hash)
                 continue
+
             simplified_summand = summand.simplify()
             if isinstance(simplified_summand, Sum):
-                summands.extend(simplified_summand.summands)
+                sum_summands = cls._simplify_summands(summands=simplified_summand.summands)
+                for op_hash, [coeff, sum_summand] in sum_summands.queue.items():
+                    new_summands.add(summand=sum_summand, coeff=coeff, op_hash=op_hash)
             else:
-                summands.append(simplified_summand)
+                new_summands.add(summand=simplified_summand)
 
-        return summands
+        return new_summands
 
     def simplify(self) -> "Sum":
-        summands = self._simplify_summands(sum_op=self)
-        return Sum(*summands)
+        new_summands = self._simplify_summands(summands=self.summands).get_summands()
+        if new_summands:
+            return Sum(*new_summands) if len(new_summands) > 1 else new_summands[0]
+        return 0
```

Comment:
Yep you are right, this might cause some issues. What do you think `qml.s_prod(0, qml.Identity))`, where `qml.Identity` is applied to all wires of the `Sum` operator?

Reply:
This way when calling `matrix()` we would get a matrix of the correct size.

Reply:
I think using `qml.s_prod(0, qml.Identity))` on all wires is a good temporary fix for now. But in the future we should add a Null operator which doesn't need to be fully initialized and can be supported with Op-arithmetic

Reply:
Done!